tests/localnet/test_localnet_codespace.py

```diff
@@ -0,0 +1,133 @@
+import platform
+from subprocess import CompletedProcess
+
+import pytest
+from pytest_httpx import HTTPXMock
+from pytest_mock import MockerFixture
+
+from tests.utils.approvals import verify
+from tests.utils.click_invoker import invoke
+from tests.utils.proc_mock import ProcMock
+
+
+def test_install_gh_already_installed(mocker: MockerFixture, proc_mock: ProcMock) -> None:
+    proc_mock.set_output(["gh", "--version"], ["some version"])
+    mocker.patch("algokit.cli.codespace.authenticate_with_github", return_value=False)
+    result = invoke("localnet codespace")
+    assert result.exit_code == 0
+
+
+def test_install_gh_not_installed_failed_install(mocker: MockerFixture, proc_mock: ProcMock) -> None:
+    proc_mock.should_fail_on(["gh", "--version"])
+    mocker.patch("algokit.cli.codespace.authenticate_with_github", return_value=False)
+    mocker.patch("algokit.core.codespace.install_github_cli_via_webi", side_effect=RuntimeError("Failed to install gh"))
+    mocker.patch("algokit.core.codespace.is_windows", side_effect=RuntimeError("Failed to install gh"))
+    result = invoke("localnet codespace")
+    assert result.exit_code == 1
+    verify(result.output)
+
+
+@pytest.mark.mock_platform_system("Windows")
+def test_install_gh_windows(
+    mocker: MockerFixture, proc_mock: ProcMock, tmp_path_factory: pytest.TempPathFactory
+) -> None:
+    cwd = tmp_path_factory.mktemp("cwd")
+    dummy_script_path = cwd / "webi_dummy_installer.ps1"
+    dummy_script_path.touch()
+
+    proc_mock.should_fail_on(
+        ["gh", "--version"],
+    )
+    proc_mock.set_output(["powershell", "--version"], ["PowerShell 7.2.1"])
```

Comment:
I don't believe that `powershell --version` is available.

Reply:
I also get the following error when the webi installer runs. It still installs though.

```
Set-ExecutionPolicy : The 'Set-ExecutionPolicy' command was found in the module 'Microsoft.PowerShell.Security', but the module could not be loaded. For more information, run 'Import-Module
```

Reply:
@neilcampbell should be resolved now, powershell is checked via different command that is more versatile. 

The Set-ExecutionPolicy on the other hand is a bit tricky to replicate, whats the output of `Get-ExecutionPolicy -List` on your side when you run it in powershell on windows (preferably same environment as the one where the error occurred during install).

Reply:
@aorumbayev Output is this:
Scope ExecutionPolicy
----- ---------------
MachinePolicy       Undefined
   UserPolicy       Undefined
      Process       Undefined
  CurrentUser       Undefined
 LocalMachine    RemoteSigned

Reply:
@aorumbayev Forgot to mention I was running `algokit -v localnet codespace`.
When running without -v it works fine without any error being printed, so we can ignore this.